packages/arb-token-bridge-ui/src/util/withdrawals/fetchWithdrawals.ts

```diff
@@ -129,36 +105,33 @@ export const fetchWithdrawals = async ({
   const currentPageStart = pageNumber * pageSize
   const currentPageEnd = currentPageStart + pageSize
 
-  const partialEthWithdrawalsFromEventLogs = [...ethWithdrawalsFromEventLogs]
-    // event logs start from the earliest, we need to reverse them
-    .reverse()
-    .slice(currentPageStart, currentPageEnd)
-  const partialTokenWithdrawalsFromEventLogs = [
-    ...tokenWithdrawalsFromEventLogs
+  // we need timestamps to sort token withdrawals along ETH withdrawals
+  const tokenWithdrawalsFromEventLogsWithTimestamp = await Promise.all(
+    tokenWithdrawalsFromEventLogs.map(withdrawal =>
+      attachTimestampToTokenWithdrawal({ withdrawal, l2Provider })
+    )
+  )
+
+  // get ETH and token withdrawals that will be displayed on the current page
+  const paginatedWithdrawalsFromEventLogs = [
+    ...ethWithdrawalsFromEventLogs,
+    ...tokenWithdrawalsFromEventLogsWithTimestamp
   ]
-    .reverse()
+    .sort((a, b) => (a.timestamp?.gt(b.timestamp || constants.Zero) ? -1 : 1))
```

Comment:
We have another instance of sorting by `timestamp in this file. Also if I'm not mistaken this is in descending orther. So let's extract to a function:

```
type Timestamped = { timestamp?: BigNumber }

function sortByTimestampDescending(a: Timestamped, b: Timestamped) {
  const aTimestamp = a.timestamp ?? constants.Zero
  const bTimestamp = b.timestamp ?? constants.Zero

 return aTimestamp.gt(bTimestamp) ? -1 : 1
}

// later
something.sort(sortByTimestampDescending)
```